src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl

```diff
@@ -787,6 +787,24 @@
     <xsl:value-of select="translate( substring-before( substring-after( concat( substring-after( $style, $property ), ';' ), ':' ), ';' ), ' ', '' )"/>
   </xsl:template>
 
+  <xsl:template name="extractTextAlignValue">
+    <xsl:param name="style"/>
+    <xsl:param name="property"/>
+    <xsl:variable name="alignValue">
+      <xsl:call-template name="extractStyleValue">
+        <xsl:with-param name="style" select="$style"/>
+        <xsl:with-param name="property" select="$property"/>
+      </xsl:call-template>
+    </xsl:variable>
+
+    <xsl:choose>
+      <xsl:when test="$alignValue = 'left' or $alignValue = 'center' or $alignValue = 'right' or $alignValue = 'justify'">
```

Comment:
Isn't there any way to iterate over alignment directions coming from `ezpublish.rng`? This will make it a bit more bulletproof whenever schema changes.

Reply:
That should be fine, as there are no other values for `text-align` attribute AFAICS and this shouldn't change anytime soon.

Reply:
> Isn't there any way to iterate over alignment directions coming from `ezpublish.rng`? This will make it a bit more bulletproof whenever schema changes.



> Maybe... It is possible to load other documents from the xsl stylesheet. However, to check such things against the schema inside the stylesheet will add a completely new level of complexity so I wouldn't recommend it.
> 
> Anyway, upon changes to the schema, new tests should be written and those should reveal this hard coded limit

I wouldn't do that. XSL is a document describing transformation from one format to another. `ezpublish.rng` is a RelaxNG schema for DocBook format. Moreover OOTB `xsl:import` should (can?) import only XSL.

It's fine as is, given this is still XSLT 1.0.